### PR TITLE
Introduction of the lifting of multiplication matrices (over the rationals)

### PR DESCRIPTION
**Pull request by mohabsafey**
* improves the lifting of rational parametrizations (saves some primes on several examples)
* implements lifting of multiplication matrices in the zero-dimensional case (called with option -L 1)
This brings some significant speedups on several families of systems, including standard benchmarks (more than 1.5 on katsura-n, more than 2 on eco-n problems, speedups increase with n). 
However, it uses more memory. This is why this option is not activated by default.

### CODE REVIEW

**Comment by ederc** [comment]
This PR also fixes #144.

File: test/diff/diff_mq_2_1.sh
[the remark this answers is not in the document]
Comment (mohabsafey) [reply test/diff/diff_mq_2_1.sh:1]: Done also, thanks

File: src/msolve/msolve.c
[the remark this answers is not in the document]
Comment (mohabsafey) [reply src/msolve/msolve.c:1]: I had difficulties with many conflicts when merging master branch with this one

**Comment by mohabsafey** [comment]
* Will add some tests for the new code
* Will split this PR so that issue https://github.com/algebraic-solving/msolve/issues/144 is fixed by another PR

File: src/msolve/main.c [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 #define DEBUGGB 0
 #define DEBUGBUILDMATRIX 0
-#define IO_DEBUG 0
+#define IO_DEBUG 1
```
[the remark this answers is not in the document]
Comment (ederc) [reply src/msolve/main.c:25]: Why did you first unset and then set it again?

File: test/diff/diff_mq_2_1.sh
Comment (ederc) [inline test/diff/diff_mq_2_1.sh:1]: Same here, you first removed the `echo` calls, then you reintroduced them.

File: src/msolve/msolve.c
[the remark this answers is not in the document]
Comment (ederc) [reply src/msolve/msolve.c:1]: What do we do about this? The current status of the PR still changes nearly 3100 lines of code in `msolve.c` mostly spaces are adjusted (I assume due to some new text editor settings?). This makes it harder to understand the origin of possible bugs found in the future.

[the remark this answers is not in the document]
Comment (mohabsafey) [reply src/msolve/msolve.c:1]: I would suggest to leave it as is. It was really a pain to fix all the merge conflicts I encountered.
What I can do is to add comments on the real changes in this file.

[the remark this answers is not in the document]
Comment (ederc) [reply src/msolve/msolve.c:1]: Well that's bad, but if you cannot resolve your git problems we have to live with it.

**Pull request merged**